packages/shared/lib/hooks/hooks.ts

```diff
@@ -27,3 +37,87 @@ export const connectionCreated = async (
 export const connectionCreationFailed = async (connection: RecentlyCreatedConnection, provider: string, activityLogId: number | null): Promise<void> => {
     await webhookService.sendAuthUpdate(connection, provider, false, activityLogId);
 };
+
+export const connectionTest = async (
+    provider: string,
+    template: ProviderTemplate,
+    credentials: ApiKeyCredentials | BasicApiCredentials,
+    connectionId: string,
+    providerConfigKey: string,
+    environment_id: number,
+    connection_config: ConnectionConfig,
+    tracer: Tracer
+): Promise<Result<boolean, NangoError>> => {
+    const active = tracer.scope().active();
+    const span = tracer.startSpan(SpanTypes.CONNECTION_TEST, {
+        childOf: active as Span
+    });
+
+    span.setTag('nango.provider', provider).setTag('nango.providerConfigKey', providerConfigKey).setTag('nango.connectionId', connectionId);
+
+    const providerVerification = template?.proxy?.verification;
+
+    if (!providerVerification) {
+        return resultOk(true);
+    }
```

Comment:
I would wrap the calls inside `apiAuth.controller.ts` instead of doing it here, that would make it clear we don't expect test when there are none; and also you have an issue with an unfinished span (or an empty one if you choose to keep this if)

Reply:
> that would make it clear we don't expect test when there are none

I'm unclear what you mean exactly here?

>  and also you have an issue with an unfinished span (or an empty one if you choose to keep this if)

Reordered the calls so the span doesn't activate if there is no test to run.

I would rather keep the span in the hook to make it more explicit and less repeatable -- in case we want to add this hook for other auth types.